### ISSUE TITLE
Allow functions to return unit by default

### ISSUE THREAD

**Issue by Kixiron** [vmware/differential-datalog#736]
For functions that return unit you have to explicitly type them as such, e.g.
```prolog
function do_something(int: u32): () { .. }
```
but not giving a return type should automatically default to unit (like Rust does), which would make this
```prolog
function do_something(int: u32) { .. }
```
Currently trying the latter gives this error
```
unexpected "{"
expecting ":"
```

**Comment by ryzhyk** [comment]
Makes sense.  Ddlog functions were initially pure (other than logging functions), but we now allow mutable arguments, so there are quite a few functions that return `()`. This is a tiny fix, I'll try to do it ASAP.

**Comment by ryzhyk** [comment]
Resolved in #756
